pyomo/repn/linear.py

```diff
@@ -857,12 +818,64 @@ def __init__(self, subexpression_cache, var_map, var_order):
         self.var_order = var_order
         self._eval_expr_visitor = _EvaluationVisitor(True)
 
+    def _eval_fixed(self, obj):
+        ans = obj.value
+        if ans.__class__ not in native_numeric_types:
```

Comment:
This is completely duplicate code to that on L#845-878 (with `obj` instead of `expr`). Worth pulling out to its own function.

Reply:
Oh, and it's duplicated in `nl_writer.py`. So even more relevant suggestion.

Reply:
I caught that and hope to further refactor the LP / NL writers to remove redundant code / logic between them (but that will be a separate PR after the next release)